README.md

```diff
@@ -1,46 +1,214 @@
+
+```python
+task_results = rpc.poll(timeout=None)
+```
+
+By default there are three files listed in `TaskResults` message: `result.json`, `stderr.log` and `stdout.log`. Result returned by `dummy_task` function is serialized to a `result.json` of form:
+
+```json
+{
+    "data": 3,
+}
+```
+
+If there are any errors in user supplied function the JSON object will contain `error` key:
+
+```json
+{
+    "error": "Some error message"
+}
+```
+
+Pick result file, load it and verify the results:
+
+```python
+import json
+
+result_json_file = None
+
+for f in task_results['results']:
+    if f.endswith('result.json'):
+        result_json_file = f
+        break
+
+with open(result_json_file, 'r') as f:
+    result_json = json.load(f)
+assert result_json['data'] == (1 + 2)
+```
+
+## Big input files
+
+By default `CreateTask` message cannot exceed 0.5MB. One must use `resources` task's field instead function `args` to supply bigger file inputs. Files listed in `resources` will be laid out in `/golem/resources` directory which is accessible from user supplied function, e.g.:
+
+```python
+def echo_task(args):
+    my_input = None
+    with open('/golem/resources/my_input.txt') as f:
+        my_input = f.read()
+    return my_input
+```
+
+is valid if followed by message:
+
+```python
+message = {
+    'type': 'CreateTask',
+    'task': {
+        'type': 'GLambda',
+        'options': {
+            'method': echo_task,
+        },
+        'resources': ['my_input.txt']
+    }
+}
+```
+
+## Custom output files
+
+There is no size limit for `result.json` file although one might want to use format different than JSON. To get back results in custom format user has to write them to a file in `/golem/output` directory. Every file in this directory will be packaged and sent back to user if it's listed in `outputs` field of `CreateTask` [message](https://github.com/golemfactory/golemrpc/blob/threaded/docs/messages.md), e.g.: 
+
+```python
+def echo_task(args):
+    with open('/golem/resources/my_input.txt', 'r') as fin,\
+         open('/golem/output/my_output.txt', 'w') as fout:
+        fout.write(fin.read())
+```
+
+will create `my_output.txt` result file and send it back to user if created by message:
+
+```python
+import os
+message = {
+    'type': 'CreateTask',
+    'task': {
+        'type': 'GLambda',
+        'options': {
+            'method': echo_task,
+            'outputs': ['my_output.txt']
+        },
+        'resources': ['{cwd}/my_input.txt'.format(cwd=os.getcwd())]
+    }
+}
+```
+
+## Local Golem nodes setup
+
+First of all, in main Golem repository `golemfactory/golem` branch `glambda0.3` required for `GLambda` tasks to work is not yet merged into develop. One has to install Golem from source using `glambda0.3` branch. See Golem instructions for [running from source](https://github.com/golemfactory/golem/wiki/Installation#running-from-the-source).
```

Comment:
We might use this opportunity and clean up the official wiki somewhat.

Reply:
Yeah, shall we add an issue to our backlog? I know this is somewhat out of scope of our team, but it's important.